app/helpers/datasets_helper.rb

```diff
@@ -1,82 +1,105 @@
 module DatasetsHelper
 
   FREQUENCIES = {
-    'annual' => { years: 1 },
-    'quarterly' => { months: 4 },
-    'monthly' => { months: 1 },
-    'daily' => { days: 1 }
+      'annual' => {years: 1},
+      'quarterly' => {months: 4},
+      'monthly' => {months: 1},
+      'daily' => {days: 1}
   }
 
   NO_MORE = {
-    'discontinued' => 'Dataset no longer updated',
-    'never' => 'No future updates',
-    'one off' => 'No future updates'
+      'discontinued' => 'Dataset no longer updated',
+      'never' => 'No future updates',
+      'one off' => 'No future updates',
+      'default' => 'Not available'
   }
 
-  def documents(datafiles)
-    docs = []
-    datafiles.each do |file|
-      if documentation?(file['documentation'])
-        docs << file['documentation']
-      end
-    end
-    docs
-  end
-
   def format(timestamp)
     Time.parse(timestamp).strftime("%d %B %Y")
   end
 
   def no_documents?(datafiles)
-    d = documents(datafiles)
-    d.count == 0
+    documents(datafiles).count == 0
   end
 
   def documentation?(key)
     key != nil && key != ""
   end
 
   def expected_update(dataset)
-    dataset['frequency'].nil? ? "Not available" : datafile_next_updated(dataset)
+    dataset['frequency'].nil? ?
+        NO_MORE['default'] :
+        datafile_next_updated(dataset)
+  end
+
+  def expected_update_class_for(freq)
+    NO_MORE.include?(freq)
+    "unavailable"
   end
 
+  def dataset_location(dataset)
+    locations(dataset).empty? ? NO_MORE['default'] : locations(dataset)
+  end
+
+  def expected_location_class_for(dataset)
+    NO_MORE['default'] if locations(dataset).empty?
+  end
+
+  def timeseries_data?(dataset)
+    dataset['datafiles'].map {|file| Time.parse(file['start_date']).year}.uniq
```

Comment:
I hadn't finished this guy, it is just returning an array of years at the mo. If it's empty or has only one date in it then it's not time series?